ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/AddBundleContainerSelectionPage.java

```diff
@@ -179,10 +179,12 @@ public boolean performFinish() {
 							if (oldContainers == null) {
 								fTarget.setTargetLocations(new ITargetLocation[] {container});
 							} else {
-								ITargetLocation[] newContainers = new ITargetLocation[oldContainers.length + 1];
-								System.arraycopy(oldContainers, 0, newContainers, 0, oldContainers.length);
-								newContainers[newContainers.length - 1] = container;
-								fTarget.setTargetLocations(newContainers);
+								if (!Arrays.asList(oldContainers).contains(container)) {
```

Comment:
If you moved the guard condition up into the `} else {` as `} else if (!Arrays.asList(oldContainers).contains(container)) {` I think the code would be cleaner and there would be fewer lines marked as changed.

Reply:
Thanks Ed ! Done.